// @tavily/n8n-nodes-tavily/dist/nodes/Tavily/actions/search/query.operation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.execute = exports.description = exports.properties = void 0;
const transport_1 = require("../../transport");
const display_1 = require("../../display");
const descriptions_1 = require("../../descriptions");
exports.properties = [
    {
        displayName: 'Query',
        name: 'query',
        description: 'Type your query',
        type: 'string',
        required: true,
        default: '',
        placeholder: 'e.g. who is leo messi?',
        displayOptions: {
            show: {
                resource: ['search'],
            },
        },
    },
    {
        displayName: 'Options',
        name: 'options',
        type: 'collection',
        placeholder: 'Add option',
        default: {},
        options: descriptions_1.queryOptions,
    },
];
const displayOptions = {
    show: {
        resource: ['search'],
        operation: ['query'],
    },
};
exports.description = (0, display_1.updateDisplayOptions)(displayOptions, exports.properties);
async function execute(index) {
    const query = this.getNodeParameter('query', index);
    const options = this.getNodeParameter('options', index);
    const body = {
        'query': query,
        ...options,
    };
    const endpoint = "/search";
    const responseData = await transport_1.tavilyApiRequest.call(this, 'POST', endpoint, body);
    return this.helpers.constructExecutionMetaData(this.helpers.returnJsonArray(responseData), { itemData: { item: index } });
}
exports.execute = execute;
