// @tavily/n8n-nodes-tavily/dist/nodes/Tavily/actions/crawl/crawl.operation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.execute = exports.description = exports.properties = void 0;
const transport_1 = require("../../transport");
const common_descriptions_1 = require("../../descriptions/common.descriptions");
const display_1 = require("../../display");
exports.properties = [
    {
        displayName: 'URL',
        name: 'url',
        type: 'string',
        required: true,
        default: '',
        description: 'The root URL to begin the crawl',
        placeholder: 'https://www.example.com',
        displayOptions: {
            show: {
                resource: ['crawl'],
            },
        },
    },
    {
        displayName: 'Options',
        name: 'options',
        type: 'collection',
        placeholder: 'Add option',
        default: {},
        options: common_descriptions_1.crawlOptions.filter(option => option.name !== 'url'),
        displayOptions: {
            show: {
                resource: ['crawl'],
            },
        },
    },
];
const displayOptions = {
    show: {
        resource: ['crawl'],
        operation: ['crawl'],
    },
};
exports.description = (0, display_1.updateDisplayOptions)(displayOptions, exports.properties);
async function execute(index) {
    const url = this.getNodeParameter('url', index);
    const options = this.getNodeParameter('options', index);
    const body = {
        url,
        ...options,
    };
    const responseData = await transport_1.tavilyApiRequest.call(this, 'POST', '/crawl', body);
    return this.helpers.returnJsonArray([responseData]);
}
exports.execute = execute;
