// @tavily/n8n-nodes-tavily/dist/nodes/Tavily/actions/node.description.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.description = void 0;
const search = __importStar(require("./search"));
const extract = __importStar(require("./extract"));
const crawl = __importStar(require("./crawl"));
exports.description = {
    displayName: 'Tavily',
    name: 'tavily',
    group: ['transform'],
    icon: 'file:img.svg',
    version: 1,
    subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
    description: 'Tavily API',
    defaults: {
        name: 'Tavily',
    },
    usableAsTool: true,
    inputs: ["main"],
    outputs: `={{['main']}}`,
    credentials: [
        {
            name: 'tavilyApi',
            required: true,
        },
    ],
    properties: [
        {
            displayName: 'Resource',
            name: 'resource',
            type: 'options',
            noDataExpression: true,
            default: 'search',
            options: [
                {
                    name: 'Search',
                    value: 'search',
                },
                {
                    name: 'Extract',
                    value: 'extract',
                },
                {
                    name: 'Crawl',
                    value: 'crawl',
                },
            ],
        },
        ...extract.description,
        ...search.description,
        ...crawl.description
    ],
};
