// @tavily/n8n-nodes-tavily/dist/credentials/TavilyApi.credentials.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TavilyApi = void 0;
class TavilyApi {
    constructor() {
        this.name = 'tavilyApi';
        this.displayName = 'Tavily API';
        this.documentationUrl = 'https://docs.tavily.com/docs/welcome';
        this.icon = 'file:icons/img.svg';
        this.properties = [
            {
                displayName: 'API Key',
                name: 'apiKey',
                type: 'string',
                typeOptions: { password: true },
                description: "Tavily API key. You can find your API key in your Tavily dashboard at https://app.tavily.com/home",
                default: "",
            },
            {
                displayName: 'Base URL',
                name: 'baseUrl',
                type: 'string',
                description: "Base URL of Tavily's API",
                default: 'https://api.tavily.com',
            },
        ];
        this.authenticate = {
            type: 'generic',
            properties: {
                headers: {
                    Authorization: '=Bearer {{$credentials.apiKey}}',
                    'X-Client-Source': 'n8n'
                },
            },
        };
        this.test = {
            request: {
                baseURL: 'https://api.tavily.com',
                url: '/search',
                method: 'POST',
                headers: {
                    Authorization: '=Bearer {{$credentials.apiKey}}',
                    'X-Client-Source': 'n8n'
                },
                body: {
                    query: 'Hello n8n!',
                },
            },
        };
    }
}
exports.TavilyApi = TavilyApi;
